### ISSUE TITLE
NVDA does not read tooltips in some applications

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3314]
**Reported by k_kolev1985 on 2013-06-30 09:42**
Even though the option to report tooltips in NVDA's options, in "Object presentation" is checked, NVDA does not read the popped up tooltips in several applications. Example of such applications are: Skype, Cobian Backup, Balabolka, KMPlayer, and others. I think the common between all those programs is that they are written in the Delphi programming language, but I may be wrong.

Steps to reproduce it:
1. Open one of the above mentioned programs.
2. Find an object (eg. a button in a toolbar) witch has a tooltip.
3. Move the mouse pointer over that object.
Expected result: NVDA should read the tooltip witch pops up for that object.
Actual result: NVDA not only does not read the tooltip witch pops up, but doesn't announce that a tooltip has appeared on the screen.

Note: I'm using NVDA version next-9303,7db4c21, but this issue is from as long as I can remember with programs like the ones mentioned above. If needs be, I'll tell you the names of the programs as well.

**Comment by bhavyashah** [comment]
@jcsteh From your https://github.com/nvaccess/nvda/issues/3314#issuecomment-155307892, I take it that addressing this issue as a whole is not possible and case-by-case patches will be required. In that case, may we close this ticket and request @k_kolev1985 to file individual tickets for each program separately? If not, am I missing something?

**Comment by jcsteh** [comment]
If some or all of these apps do all indeed use a common toolkit, it's possible a single patch might cover multiple apps. We'd need the window class names for tooltips in all affected apps. Unfortunately, this is pretty difficult to get without watching for show events in an event tracing tool like AccEvent or AccProbe.

**Comment by bhavyashah** [comment]
CC @k_kolev1985 to keep https://github.com/nvaccess/nvda/issues/3314#issuecomment-322168012 in mind...

**Comment by Adriani90** [comment]
cc: @k-kolev1985 I guess this is still reproducible with NVDA 2019.1.1 in most of the applications described above. Is this correct?

**Comment by k-kolev1985** [comment]
@Adriani90 Yes, this is still reproducible at least with Cobian Backup, Balabolka. The mentioned Skype example was before version 8, which was a totally different thing - now the tooltips in Skype 8 are read by NVDA. I don't have a recent version of KMPlayer to test it out on it (I don't use it anymore).

**Comment by Adriani90** [comment]
I think this needs to be fixed by the app developer, like it was fixed in Skype 8. I am adding needs external fix here.

**Comment by Adriani90** [comment]
Oh wrong button :)

**Comment by Adriani90** [comment]
We didn't get any updates if this is still an issue in Balabolka, KMPlayer or similar application. In Skype, most of the tooltips work, but not all of them. This seems an issue which needs to be discussed with the app developers. They have to expose the tooltips or notifications or allerts correctly via UIA or another accessibility API to make it possible for NVDA to access them.

I think NV Access can decide better if this issue still needs to be tracked here or not.

**Comment by k-kolev1985** [comment]
@Adriani90 The tooltips are still not read in Cobian Backup 11 and the latest version of Balabolka.
As I said in my previous comment here - I don't use KMPlayer anymore and therefore can't say about it.
And as we both said previously - now the tooltips in Skype are read (or at least the ones I came across are anyway).
